app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/uploadsupportingdocs/UploadSupportingEvidenceController.scala

```diff
@@ -0,0 +1,1012 @@
+        val result = for {
+          _ <- upscanService.removeFile(DraftReturnId(draftReturnId.toString), UpscanInitiateReference(id))
+          updatedAnswers = answers match {
+            case IncompleteUploadSupportingEvidenceAnswers(
+                doYouWantToUploadSupportingEvidenceAnswer,
+                supportingEvidences
+                ) =>
+              IncompleteUploadSupportingEvidenceAnswers(
+                doYouWantToUploadSupportingEvidenceAnswer,
+                supportingEvidences.filterNot(c => c.reference == id)
+              )
+            case CompleteUploadSupportingEvidenceAnswers(
+                doYouWantToUploadSupportingEvidenceAnswer,
+                supportingEvidences
+                ) =>
+              CompleteUploadSupportingEvidenceAnswers(
+                doYouWantToUploadSupportingEvidenceAnswer,
+                supportingEvidences.filterNot(c => c.reference == id)
+              )
+          }
+          newDraftReturn = fillingOutReturn.draftReturn match {
+            case s @ DraftSingleDisposalReturn(
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _
+                ) =>
+              s.copy(uploadSupportingDocuments = Some(updatedAnswers))
+            case m @ DraftMultipleDisposalsReturn(
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _
+                ) =>
+              m.copy(uploadSupportingDocuments = Some(updatedAnswers))
+          }
+
+          _ <- returnsService
+                .storeDraftReturn(
+                  newDraftReturn,
+                  fillingOutReturn.subscribedDetails.cgtReference,
+                  fillingOutReturn.agentReferenceNumber
+                )
+          _ <- EitherT(
+                updateSession(sessionStore, request)(
+                  _.copy(journeyStatus = Some(fillingOutReturn.copy(draftReturn = newDraftReturn)))
+                )
+              )
+        } yield ()
+
+        result.fold({ e =>
+          logger.warn("Could not update session", e)
+          errorHandler.errorResult()
+        }, _ => Redirect(routes.UploadSupportingEvidenceController.checkYourAnswers()))
+      }
+  }
+
+  def checkYourAnswers(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    withUploadSupportingEvidenceAnswers(request) { (_, _, _, fillingOutReturn, answers) =>
+      (answers, fillingOutReturn.draftReturn) match {
+        case (c: UploadSupportingEvidenceAnswers, s: DraftSingleDisposalReturn) => {
+          checkYourAnswersHandler(c, fillingOutReturn, s)
+        }
+        case (c: UploadSupportingEvidenceAnswers, s: DraftMultipleDisposalsReturn) =>
```

Comment:
there's no point having separate case statements for single and multiple draft returns - they both point to the same function which takes in a generic `DraftReturn` anyway